src/web/components/StickyBottomBanner/BrazeBanner.tsx

```diff
@@ -154,6 +155,9 @@ export const canShow = async (
     asyncBrazeUuid: Promise<null | string>,
     isDigitalSubscriber: undefined | boolean,
 ): Promise<CanShowResult> => {
+    const canShowTiming = initPerf('braze-banner');
```

Comment:
This is fine, but if we find it useful, I wonder if it would be nice to wrap it around the can show logic for all banners? Out of scope for this PR and let's see how we find it.

Reply:
I'm not crazy about how the performance logic is all intertwined with the business logic of the Braze banner. If we wanted to tidy things now would we be able to wrap the business logic promise in timer functionality?

Something like:

    time(config params e.g. logging / ophan record data, <promise to time>)

If such a `time` function accepted and returned a promise then it can be kept separate from business logic by wrapping. Such a function would probably be useful in a separate (re-usable) module, subject to the same questions above about purpose and audience (and thus, side effects).

```typescript
const canShowLogic = async (
    asyncBrazeUuid: Promise<null | string>,
    isDigitalSubscriber: undefined | boolean,
): Promise<CanShowResult> => {
  // Braze logic here as before
}

export const canShow = time(logging = false, canShowLogic)
```

I don't feel strongly here though, very happy for this to go out as-is for us to investigate. We can look at how future use-cases appear and decide whether there's a good abstraction to address those in the future when we have a better view of how it should work.

Reply:
That sounds sensible - I know with the current structure it's much simpler to put the timing within the BrazeBanner itself but it is probably worth revisiting in a future PR if we want to monitor their performance in the future.